### PR TITLE
GH-14152: Add "done" button for read only mode

### LINKED ISSUE #14152: Change "Add to Cart" Button component that in read only (order configuration) the buttons says done and navigates to order history.
Change "Add to Cart" Button component that in read only (order configuration) the buttons says done and navigates to order history.
Currently the button is not shown in an order configuration.
The button needs to be shown with label "done".

See wires (at the bottom):
https://4prsat.axshare.com/#id=0dc98d&p=d_summary_from_order_history&g=1

### PR DESCRIPTION
**Pull request by Changsuwan**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.spec.ts [hunk 1]
```diff
@@ -356,4 +428,24 @@ describe('ConfigAddToCartButtonComponent', () => {
       expect(globalMessageService.add).toHaveBeenCalledTimes(0);
     });
   });
+  describe('displayOnlyButton', () => {
+    it('should navigate to order overview', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.spec.ts]:
```suggestion
    it('should navigate to review order', () => {
```

File: feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts [hunk 2]
```diff
@@ -223,4 +230,34 @@ export class ConfiguratorAddToCartButtonComponent {
...
       });
   }
+  leaveConfigurationOverview(): void {
+    this.container$.pipe(take(1)).subscribe((container) => {
+      if (
+        container.routerData.owner.type ===
+        CommonConfigurator.OwnerType.ORDER_ENTRY
+      ) {
+        this.goToOrderDetails(container.routerData.owner);
+      } else if (
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]: do we need this if statement? Because we can only have 2 cases here. DisplayOnly is only called for order history or review order

File: feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts [hunk 3]
```diff
@@ -223,4 +230,34 @@ export class ConfiguratorAddToCartButtonComponent {
...
+        container.routerData.owner.type ===
+        CommonConfigurator.OwnerType.CART_ENTRY
+      ) {
+        this.routingService.go({ cxRoute: 'checkoutReviewOrder' });
+      }
+    });
+  }
+
+  private goToOrderDetails(owner: CommonConfigurator.Owner): void {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]:
```suggestion
  protected goToOrderDetails(owner: CommonConfigurator.Owner): void {
```

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts (hunk 2 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]: Will we only have 2 cases? If yes, that's a valid point. But what about other processes where you can view the configuration as well without being able to change it (e.g. Return Process)? Otherwise, I think we should keep the method this way.

Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]: We can never change the configuration for an order, therefore I don't understand your comment...
From my point of view: We only have those 2 cases

Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]: Ok, done.

**Review by ChristophHi — APPROVED** [review]

File: feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts [hunk 4]
```diff
@@ -54,7 +58,10 @@ export class ConfiguratorAddToCartButtonComponent {
     protected configuratorCartService: ConfiguratorCartService,
     protected configuratorGroupsService: ConfiguratorGroupsService,
     protected configRouterExtractorService: ConfiguratorRouterExtractorService,
-    protected globalMessageService: GlobalMessageService
+    protected globalMessageService: GlobalMessageService,
+    protected userOrderService: OrderFacade,
+    protected commonConfiguratorUtilsService: CommonConfiguratorUtilsService,
+    protected windowRef: WindowRef
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]: Why is that needed? Please remove...

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts (hunk 4 above)
Comment (Changsuwan) [reply feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts]: Done

**Review by ChristophHi — DISMISSED** [review]

**Review by ChristophHi — APPROVED** [review]

**Review by Platonn — APPROVED** [review]
